atomics/include/desul/atomics/cuda/cuda_cc9_asm_exchange_op.inc

```diff
@@ -0,0 +1,25 @@
+
+#define __DESUL_IMPL_CUDA_ASM_ATOMIC_EXCHANGE() \
+template<class ctype> \
+inline __device__ typename ::std::enable_if<sizeof(ctype)==16, ctype>::type device_atomic_exchange(ctype* dest, ctype value, __DESUL_IMPL_CUDA_ASM_MEMORY_ORDER, __DESUL_IMPL_CUDA_ASM_MEMORY_SCOPE) { \
+  __int128_t asm_value = reinterpret_cast<__int128_t&>(value); \
```

Comment:
Have you seen https://docs.nvidia.com/cuda/cuda-c-programming-guide/#nv-atomic-exchange ?

Reply:
same for this